source/base/function_signed_distance.cc

```diff
@@ -332,6 +332,88 @@ namespace Functions
 
       return evaluate_ellipsoid(point) < 0.0 ? -distance : distance;
     }
+
+
+
+    template <int dim>
+    Rectangle<dim>::Rectangle(const Point<dim> &bottom_left,
+                              const Point<dim> &top_right)
+      : bounding_box({bottom_left, top_right})
+    {}
+
+
+
+    template <int dim>
+    double
+    Rectangle<dim>::value(const Point<dim> & p,
+                          const unsigned int component) const
+    {
+      (void)component;
+
+      // calculate signed distance vector to bounding box
+      std::array<double, dim> distances;
+      for (unsigned int d = 0; d < dim; ++d)
+        distances[d] = bounding_box.signed_distance(p, d);
+
+      if (dim == 1)
+        {
+          // case: specialization for 1D
+          return distances[0];
+        }
+
+      unsigned int counter = 0;
+
+      for (unsigned int d = 0; d < dim; ++d)
+        if (distances[d] > 0)
+          counter++;
+
+      if (counter == 0)
```

Comment:
This took me far too long to understand, primarily because `counter` didn't give away what it is you are counting. Can you name the variable `n_directions_outside` or some such?

Reply:
You are absolutely right!